src/veryfi/Client.php

```diff
@@ -550,4 +551,297 @@ public function replace_tags(int $document_id,
         return $this->request('PUT', $endpoint_name, $request_arguments);
     }
 
+    /**
+     * Get all W2 documents.
+     *
+     * @param int|null $page The page number, response is capped to a maximum of 50 results per page.
+     * @return string An array of JSON with all W2 documents.
+     * @throws Exception when API version is not supported for W2 documents.
+     */
+    public function get_w2_documents(int $page = null): string
+    {
+        $endpoint_name = '/w2s/';
+        $params = [];
+        if ($page !== null) {
+            $params['page'] = $page;
+        }
+        return $this->request('GET', $endpoint_name, $params);
+    }
+
+    /**
+     * Get a W2 document.
+     *
+     * @param string $document_id The ID of the document you'd like to retrieve.
+     * @return string Data extracted from the document.
+     */
+    public function get_w2_document(string $document_id): string
+    {
+        $endpoint_name = "/w2s/{$document_id}/";
+        $request_arguments = ['id' => $document_id];
+        return $this->request('GET', $endpoint_name, $request_arguments);
+    }
+
+    /**
+     * Process a W2 document from a file path and extract all fields from it.
+     *
+     * @param string $file_path Path on disk to a file to submit for data extraction.
+     * @param bool $delete_after_processing Delete this document from Veryfi after data has been extracted.
+     * @param int $max_pages_to_process The number of pages to process for the document.
+     * @param array $additional_request_parameters Additional request parameters.
+     * @return string Data extracted from the document.
+     */
+    public function process_w2_document(string $file_path,
+                                        bool $delete_after_processing = false,
```

Comment:
let's use the same naming as in API - auto_delete

Reply:
Done